test/bin/manage_build_cache.sh

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+#
+# This script should be run on the build host to manage cache of build artifacts
+set -euo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# shellcheck source=test/bin/common.sh
+source "${SCRIPTDIR}/common.sh"
+
+AWS_BUCKET_NAME="${AWS_BUCKET_NAME:-microshift-build-cache}"
+RELEASE_SUBDIR=
+BUILDID_SUBDIR=
+
+usage() {
+    cat - <<EOF
+${BASH_SOURCE[0]} (upload|download|exist|latest|keep) [options]
+
+Manage build cache at the '${AWS_BUCKET_NAME}' AWS S3 bucket. The script
+assumes that the bucket exists and it is configured for read-write access
+using 'aws s3 (ls|sync|rm)' operations.
+
+The cache directory structure is '${AWS_BUCKET_NAME}/<release>/<buildid>'.
+
+  -h        Show this help.
+
+  upload:   Upload build artifacts from the local disk to the specified
+            '${AWS_BUCKET_NAME}/<release>/<buildid>' AWS S3 bucket.
+
+  download: Download build artifacts from the specified
+            '${AWS_BUCKET_NAME}/<release>/<buildid>' AWS S3 bucket to
+            the local disk.
+
+  exist:    Exit with 0 status if the specified
+            '${AWS_BUCKET_NAME}/<release>/<buildid>' sub-directory exists
+            in a AWS S3 bucket, 1 otherwise.
+
+  latest:   Update the '${AWS_BUCKET_NAME}/<release>/latest' directory
+            in the AWS S3 bucket to a copy of the specified build ID.
+
+  keep:     Delete all data from the '${AWS_BUCKET_NAME}/<release>'
+            AWS S3 bucket, only keeping the '<buildid>' sub-directory.
+
+Options:
+
+  -r <release>  The release sub-directory in the cache.
+
+  -b <buildid>  The build ID sub-directory in the cache.
+
+EOF
+}
+
+action_upload() {
+    local -r src_dir="${IMAGEDIR}/${VM_POOL_BASENAME}"
```

Comment:
`VM_POOL_BASENAME` suggests we only cache iso?

Reply:
At the moment, yes. I thougth about this - if we need to expand to commits, we can add third mandatory option which will switch between pool and ostree repo dirs.

Reply:
What ISOs would we cache right now? rhel92.image-installer and what else?

I think it would be nice to cache commits like current (4.15-ec) or previous (4.14-rc) release, they're pretty much static (i.e. updating nightly should be good enough).

Reply:
Also, we could have commits based of `main` cache using post-submit job

Reply:
I'm planning to cache group1 and group2, so we'll get 2 ISO's.
Let's see how this works and proceed to commits later?

Reply:
Why do we have any ISOs in group2?

Can we put the cacheable images into their own group(s) so it's clear what items can/should be cached? Maybe use `cachedN` as the pattern for directory names?

Reply:
Yes, we can certainly reorg the structure further. At this time, however, we can start by implementing some form of caching that will evolve in the future and dictate new structure of dirs, etc.